### PR TITLE
MM-43667 - start trial from invite guest user

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
This PR adds the logic to prevent users to invite guest unless they are in free trial or paid features. Also display the call to action and indicator letting admins know that inviting guests is a restricted feature.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-43667

#### Related Pull Requests
n/a

#### Screenshots
Captura de pantalla 2022-06-21 a las 13 20 56
Captura de pantalla 2022-06-21 a las 13 54 23
Captura de pantalla 2022-06-22 a las 10 45 13

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by pvev** [comment]
@annelieseklein @marianunez the [current design for end users](https://www.figma.com/file/Aob5ffHtfjKzQoebR93ycA/Cloud-Freemium?node-id=3617%3A222755)  (which is not yet implemented, this PR takes care only to the feature restriction to admins) shows in the action modal a "back" button, which if clicked should open back the Invite people modal. In this implementation the modal is showing the CTA to start the trial and as a secondary button the "View Plans" CTA button. My question is, for admins do we want to keep it this way? or we want admins to also have the "Back" button which will open again the "Invite People" modal?

Captura de pantalla 2022-06-21 a las 13 21 21

File: components/common/radio_group.tsx [hunk 1]
```diff
@@ -57,8 +57,8 @@ const RadioButtonGroup: React.FC<RadioGroupProps> = ({
                     }
                 </label>
                 {(badge && val === badge?.matchVal) &&
-                    <Badge className='radio-badge'>
-                        {badge.text}
+                    <Badge className={`radio-badge ${badge.extraClass ?? ''}`}>
```
Comment (julmondragon) [inline components/common/radio_group.tsx:60]: classNames ?

File: components/invitation_modal/invite_as.tsx [hunk 2]
```diff
@@ -49,24 +135,26 @@ export default function InviteAs(props: Props) {
                         },
                         {
                             key: (
-                                <span className='InviteAs__label'>
+                                <span className={`InviteAs__label ${guestDisabledClass}`}>
```
Comment (julmondragon) [inline components/invitation_modal/invite_as.tsx:138]: classNames ?

File: components/invitation_modal/invite_as.tsx [hunk 3]
```diff
@@ -23,6 +40,75 @@ export type Props = {
...
+            matchVal: InviteType.GUEST as string,
+            badgeContent: restrictedIndicator,
+            extraClass: 'Badge__restricted-indicator-badge',
+        };
+        extraGuestLegend = false;
+    }
+
+    // disable the radio button logic (is disabled when is cloud starter - pre and post trial)
+    if (isCloud && ((!isCloudFreeTrial && !isPaidSubscription) || (hasPriorTrial && !isPaidSubscription))) {
+        guestDisabled = (id: string) => {
```
Comment (julmondragon) [inline components/invitation_modal/invite_as.tsx:107]: Could this be a const? The if condition would work internally as well, right?

File: components/invitation_modal/invite_as.tsx [hunk 4]
```diff
@@ -23,6 +40,75 @@ export type Props = {
...
+            clickCallback={closeInviteModal}
+        />
+    );
+
+    // show the badge logic (the restricted indicator takes care of the look when it is trial or not)
+    if (isSystemAdmin && isCloud && !isPaidSubscription) {
+        guestDisabledClass = isCloudFreeTrial ? '' : 'disabled-legend';
+        badges = {
+            matchVal: InviteType.GUEST as string,
+            badgeContent: restrictedIndicator,
```
Comment (julmondragon) [inline components/invitation_modal/invite_as.tsx:99]: Defining the indicator inline here would ensure it's only created when actually required, right?

**Review by julmondragon — APPROVED** [review]
Looks good!

File: components/invitation_modal/invite_as.tsx (hunk 2 above)
Comment (pvev) [reply components/invitation_modal/invite_as.tsx:138]: as personal preference, I rather to keep it simple using string literals if it is just one single conditional class.

File: components/invitation_modal/invite_as.tsx (hunk 4 above)
Comment (pvev) [reply components/invitation_modal/invite_as.tsx:99]: Agree! my only concern is that the code will be harder to read.

**Comment by annelieseklein** [comment]
> > @pvev
> > 
> > 1. Can we make the "Copy Invite link" on the modal a secondary button please?
> > 2. Can we please change the copy for the modal to say "Invite people" instead of "Invite members"
> 
> @annelieseklein are you referring to this modal? Captura de pantalla 2022-06-22 a las 17 57 24

Yup!

**Comment by pvev** [comment]
> > > @pvev
> > > 
> > > 1. Can we make the "Copy Invite link" on the modal a secondary button please?
> > > 2. Can we please change the copy for the modal to say "Invite people" instead of "Invite members"
> > 
> > 
> > @annelieseklein are you referring to this modal? Captura de pantalla 2022-06-22 a las 17 57 24
> 
> Yup!

ok, thanks!

**Comment by pvev** [comment]
@annelieseklein @stevemudie friendly reminder on this one :)

**Review by stevemudie — APPROVED** [review]
All set @pvev 
I stepped it through all of it's paces from new Workspace, to trial and then post trial and it all worked as expected!

**Comment by annelieseklein** [comment]
LGTM!

Small thing, on the tooltip description can we please change "paid" to "professional"? Thanks! Great job!

**Pull request merged**